### DISCUSSION TITLE
Does anyone configure CppAD on Visual Studio? I found some confusing problems in there

### DISCUSSION THREAD

**Discussion by Wang-Peng-debug** [coin-or/CppAD#213] [Q&A]
![image](https://github.com/user-attachments/assets/6cf68413-7f37-40f8-b8ef-7bd10a1d856d)
Look at the color of static functions min and max, both have acutally already been defined in std, but the author defined them with the same names in the class, so the errors occurr while running. How could I fix it?

**Comment by bradbell** [comment]
I am away from the internet for a week or so and sorry I am not much help right now. Try looking at the workflows that run on windows:
https://github.com/coin-or/CppAD/tree/master/.github/workflows

**Reply by Wang-Peng-debug** [reply]
Thank you bro!!!!!!

**Reply by Wang-Peng-debug** [reply]
![image](https://github.com/user-attachments/assets/4a540bfb-4cff-468b-9b8e-8213f0e9b629)
Sorry my bro, I have no idea what they are used for and how should I use them? Could you please give me some ideas?

**Comment by bradbell** [comment]
Are you using the visual studio  version of the dos command window to run the cmake to configure CppAD ?

**Reply by Wang-Peng-debug** [reply]
I did use Cmake to generate VS project file, but didn't use dos command window. The VS project file is made, but it fails to generate solution. Anyway, I still found .lib file in the project filie after cmaking. But, the errors, which displays as " Unexpected Type "Void"" or "Grammar Errors",  still occurred while running this,  but I found no any such errors in these codes

**Reply by Wang-Peng-debug** [reply]
![image](https://github.com/user-attachments/assets/f87fd3b5-5e9b-49c0-b5a5-347a1a573858)
I took a screenshot of these errors for you, and can you read Chinese?
![image](https://github.com/user-attachments/assets/006eefc5-61e9-4a40-b392-7dab10bb4ba8)
I still took a screenshot where the errors lie in

**Comment by bradbell** [comment]
I do not know any Chinese. I suggest you wail until I am able to access a windows computer (about 10 days from now) so I can send you the exact commands for checking CppAD on your system.

**Reply by Wang-Peng-debug** [reply]
![image](https://github.com/user-attachments/assets/eb7abe61-d044-4439-8415-0d07e185799a)
Bro, are you the author of these codes? If you really are, that would be much better to discuss some issues! And now I have got those problems I mentioned before solved, but there still remains one, which is that the exception was triggered while I ran get_started.cpp, which I thought was caused by lacking of DLL file, so How do you think that?

**Comment by bradbell** [comment]
Did you run ?
```
make check_example_get_started
```

See  https://cppad.readthedocs.io/latest/cmake_check.html

**Reply by Wang-Peng-debug** [reply]
No, I haven't , but I'll give it a try. Now I am confronted with another problem, which is that an exception is triggered when the codes run here as follosws:
![image](https://github.com/user-attachments/assets/1a49b5c2-5151-420b-a657-0ef35a28a6ca)
The exception is shown as follows:
![image](https://github.com/user-attachments/assets/1b114720-8b94-4726-a4f9-fc10191e7170)
How to fix this problem?

**Reply by Wang-Peng-debug** [reply]
![image](https://github.com/user-attachments/assets/3fd98869-5d2c-4d33-8491-4bba1ffe8ce6)
Is this an operation command for Linux? It doesn't work on my Windows.

**Comment by bradbell** [comment]
This will have to wait until I return to normal internet connection. I am currently in the mountains, without my windows machine, and have a very poor connection.
